public/notification-service-worker.js

```diff
@@ -0,0 +1,64 @@
+self.addEventListener(
```

Comment:
This should be ported to ES3 for browser compatibility.

Reply:
Can you elaborate?

Reply:
Because this is in public, it won’t be covered by Babel.  Things like template literals aren’t supported in some older browsers, so it will most likely be better to just use string concatenation for it.

Reply:
That should be fine, this service worker is added only in browsers with `PushManager` support which are pretty new: https://developer.mozilla.org/en-US/docs/Web/API/PushManager#Browser_compatibility

For example, here is compatibility for string literals: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Browser_compatibility

Reply:
That's fair.  One thing I will say though is that this should in my opinion be opt-in.  Sometimes, people just don't want distractions like this.  I can open a PR to create a cookie-based configuration system if you want.

Reply:
It will be an opt-in, your browser will ask if you'll want notifications and only if you'll grant permissions you'll get the service worker and notifications.

Reply:
Ah.